### ISSUE TITLE
[Bug] Embedding manager confuses files with the same name but from different datasets

### ISSUE THREAD

**Issue by erogol** [coqui-ai/TTS#1722]
### Describe the bug

Since we key embeddings with the file name, the same file names from the different datasets would collide. 

### To Reproduce

Run the embedding manager with 2 versions of the same dataset. 

### Expected behavior

Options I see

- Use full path
- Append dataset name
- Key-based on file content

### Environment

```shell
No need for this issue
```

**Comment by tieincred** [comment]
Yes, I have had same problem I had to rename the all files in different folders to get it right, ended up wasting 32 hrs of training.

**Comment by manmay-nakhashi** [comment]
faced same problem. +1

**Comment by WeberJulian** [comment]
Maybe instead of using the filename, we can use a quick hash of the audio file instead as a key in th dict? Not sure about the performence impact though... Or instead of filename, we can use the full path as a key? This solution would break the `--old_file` argument across systems.

**Comment by erogol** [comment]
we can use ```DATASET_NAME/path/to/wav/file/file.wav as the key```. Dataset name comes from the formatted and the rest is the relative path to the root dataset folder.

**Comment by WeberJulian** [comment]
@erogol the problem is that you can use many times in a config file the same formatter (so DATASET_NAME right?) for different datasets.

**Comment by erogol** [comment]
@WeberJulian 

We can make the dataset name required argument or detect multiple uses of the same formatter and ask for a different name for the latter uses. 

The problem with content-based hash, it might change based on data preprocessing like resampling
The problem with attribute-based hash, it might change on a different computer.

**Comment by WeberJulian** [comment]
Currently, BaseDatasetConfig.name corresponds to the name of the formatter. Maybe we can add an id filed that is auto filled with  formatter name and if there is a conlict then {name}-{i}?

**Comment by erogol** [comment]
I've actually added formatters to return root_path in one of my branches. I can make a PR for it if it helps.
